pkg/vpcmodel/semanticDiffSubnets.go

```diff
@@ -0,0 +1,221 @@
+	subnet2Subtract1 := subnetConfigConnectivity2.SubnetConnectivitySubtract(&subnetConfigConnectivity1)
+
+	// 3. ToDo: grouping, see comment at the end of this file
+
+	res := &diffBetweenSubnets{
+		subnet1Subtract2: subnet1Subtract2,
+		subnet2Subtract1: subnet2Subtract1}
+	return res, nil
+}
+
+// for a given EndpointElem (representing a subnet or an external ip) in config return the EndpointElem representing the
+// subnet/external address in otherConfig or nil if the subnet does not exist in the other config.
+// ToDo: this is done based on names only at the moment. Perhaps take into account other factors such as cidr?
+// ToDo: instead of performing this search each time, use a map created once
+func (c *CloudConfig) getEndpointElemInOtherConfig(other *CloudConfig, ep EndpointElem) EndpointElem {
+	if ep.IsExternal() {
+		for _, node := range other.Nodes {
+			if node.Name() == ep.Name() {
+				res := EndpointElem(node)
+				return res
+			}
+		}
+	} else {
+		for _, nodeSet := range other.NodeSets {
+			if nodeSet.Name() == ep.Name() {
+				res := EndpointElem(nodeSet)
+				return res
+			}
+		}
+	}
+	return nil
+}
+
+// generates from subnet1Connectivity.AllowedConnsCombined and subnet2Connectivity.AllowedConnsCombined
+// Two equivalent SubnetConnectivityMap objects s.t. any (src1, dst1) of subnet1Connectivity and
+// (src2, dst2) of subnet2Connectivity are either:
+//  1. src1 disjoint src2 or dst1 disjoint dst2
+//  2. src1 = src2 and dst1 = dst2
+//     What is done here is repartitioning the ipBlocks so that the above will hold
+//
+// todo: verify that the returns objects indeed have exactly the same ipBlocks
+func (connectivity SubnetConnectivityMap) getConnectivesWithSameIPBlocks(other SubnetConnectivityMap) (
+	alignedConnectivity SubnetConnectivityMap, alignedOther SubnetConnectivityMap) {
+	// todo: use DisjointIPBlocks(set1, set2 []*IPBlock) []*IPBlock  of ipBlock.go
+	alignedConnectivity = connectivity
+	alignedOther = other
+	return
+}
+
+// SubnetConnectivitySubtract Subtract one SubnetConnectivityMap from the other
+// assumption: any connection from connectivity and "other" have src (dst) which are either disjoint or equal
+func (subnetConfConnectivity *SubnetConfigConnectivity) SubnetConnectivitySubtract(other *SubnetConfigConnectivity) SubnetsDiff {
+	connectivitySubtract := map[EndpointElem]map[EndpointElem]*connectionDiff{}
+	for src, endpointConns := range subnetConfConnectivity.subnetConnectivity {
+		for dst, conns := range endpointConns {
+			if conns.IsEmpty() {
+				continue
+			}
+			if _, ok := connectivitySubtract[src]; !ok {
+				connectivitySubtract[src] = map[EndpointElem]*connectionDiff{}
+			}
+			diffConnectionWithType := &connectionDiff{nil, NoDiff}
+			srcInOther := subnetConfConnectivity.config.getEndpointElemInOtherConfig(other.config, src)
+			dstInOther := subnetConfConnectivity.config.getEndpointElemInOtherConfig(other.config, dst)
```

Comment:
if `src` or `dst` is an external ip-range, it could be a one that originally was a wider range and then got split for alignment of ip-ranges for both connectivity-maps, so it may be an issue to search for the exact same ip-range in the config nodes..

Reply:
Sure, but this part is not implemented yet, as I mentioned above.
This is why it does not have a unit test either, yet.
Please see documentation to the not yet written function `getComparableConnections` (I'm working on it for the next PR)

Reply:
Inserted changed connection unit test

Reply:
OK, just thought the current code is a bit misleading, maybe add a comment that this should be changed and the ip-range cannot be necessarily found as-is in the config.

Reply:
I mentioned it above, and the code is not yet connected to the command line and thus can not actually be used. I'm working in the ip-range now. Comment also added.